include/fiction/algorithms/simulation/sidb/critical_temperature.hpp

```diff
@@ -165,188 +140,137 @@ template <typename Lyt>
-                    output_bits.push_back(kitty::get_bit(parameter.truth_table, parameter.input_bit) != 0u);
+                    this->determine_critical_temperature(energy_state_type);
                 }
+
                 else
                 {
-                    output_bits_index = {-4, -3};  // fo2.
-                    // Truth table entries for given input are collected.
-                    output_bits.push_back(kitty::get_bit(parameter.truth_table, parameter.input_bit * 2 + 1) != 0u);
-                    output_bits.push_back(kitty::get_bit(parameter.truth_table, parameter.input_bit * 2) != 0u);
+                    stats.critical_temperature = 0.0;  // If no ground state fulfills the logic, the Critical
+                                                       // Temperature is zero. May be worth it to change µ_.
                 }
             }
-
-            // Output cell(s) is/are collected.
-            std::vector<typename Lyt::cell> output_cells;
-            output_cells.reserve(output_bits_index.size());
-
-            // Collect the SiDB cells where the output bit is read out.
-            for (const auto& output_bit : output_bits_index)
-            {
-                output_cells.push_back(
-                    all_cells[static_cast<uint64_t>(static_cast<int64_t>(layout.num_cells()) + output_bit)]);
-            }
-
-            // A label that indicates whether the state still fulfills the logic.
-            sidb_energy_and_state_type energy_state_type{};
-            energy_state_type = calculate_energy_and_state_type(distribution, simulation_results.charge_distributions,
-                                                                output_cells, output_bits);
-
-            const auto min_energy = energy_state_type.cbegin()->first;
-
-            auto ground_state_is_transparent =
-                energy_between_ground_state_and_first_erroneous(energy_state_type, min_energy);
-
-            if (ground_state_is_transparent)
-            {
-                this->determine_critical_temperature(energy_state_type);
-            }
-
-            else
-            {
-                temperature_stats.critical_temperature = 0.0;  // If no ground state fulfills the logic, the Critical
-                                                               // Temperature is zero. May be worth it to change µ_.
-            }
         }
 
         return true;
     }
 
-    bool non_gate_based_simulation()
+    /**
+     * *Gate-based Critical Temperature* Simulation of a SiDB layout for a given Boolean function.
+     *
+     * @tparam TT The type of the truth table specifying the gate behavior.
+     * @param tt Expected Boolean function of the layout given as a multi-output truth table.
+     * @return True if the simulation succeeds and *Critical Temperature* is determined, false otherwise.
+     */
+    bool non_gate_based_simulation() noexcept
```

Comment:
```suggestion
     * @return `true` if the simulation succeeds and the *Critical Temperature* is determined, `false` otherwise.
     */
    [[nodiscard]] bool non_gate_based_simulation() noexcept
```